### ISSUE TITLE
Unexpected behavior for kinD nodes when using the --node-selector option

### ISSUE THREAD

**Issue by simcod** [srl-labs/containerlab#2106]
To reproduce this issue, the following [codebase](https://containerlab.dev/manual/kinds/k8s-kind/#kubernetes-in-docker-kind-cluster) can be used.
The creation of the kinD clusters can be used in combination with the `--node-selector` option:

```bash
sudo containerlab deploy --topo topo.yaml --node-filter k01
```

However, destruction is not possible:

```bash
sudo containerlab destroy --topo topo.yaml --node-filter k01
```

```bash
# Output
INFO[0000] no containerlab containers found
```

The expected behaviour would be that containerlab only destroys the node `k01`.

The following commands are working as expected for the node `srl01`:

```bash
sudo containerlab deploy --topo topo.yaml --node-filter srl01

sudo containerlab destroy --topo topo.yaml --node-filter srl01
```

**Comment by hellt** [comment]
Yes, node filters feature has some issues, and I presume mostly with nodes deployed outside of containerlab (like kind nodes).

**Comment by simcod** [comment]
@hellt Thanks for your quick reply.
